components/brave_ads/browser/ads_tab_helper.cc

```diff
@@ -57,11 +57,31 @@ AdsTabHelper::~AdsTabHelper() {
 
 void AdsTabHelper::DidFinishNavigation(
```

Comment:
just sidenotes:
 * why don't we filter internal and other useless urls ('chrome://*`, `file://`, IPs etc) here?
 * If `tab_id_` recorded in the constructor was invalid, shouldn't we do an early return in this method?

Reply:
We try to keep as much business logic as possible in the ads lib such as checking if a url has a http/https scheme so that iOS runs under the same logic

Reply:
Added early return, thanks